Rename some functions to match the style in rcl

davwan01 opened the pull request:
Connects to #121

davwan01 commented:
CI:

- Linux:
  - [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=2502)](http://ci.ros2.org/job/ci_linux/2502/)
- Linux-aarch64:
  - [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-aarch64&build=97)](http://ci.ros2.org/job/ci_linux-aarch64/97/)
- macOS:
  - [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_osx&build=1970)](http://ci.ros2.org/job/ci_osx/1970/)
- Windows:
  - [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=2615)](http://ci.ros2.org/job/ci_windows/2615/)

wjwwood commented:
@dirk-thomas FYI this would affect your time based logging filter I think. It should just be a find-replace, sorry for the inconvenience. I normally would have waited to do this until after your stuff landed, but it was a decent simple task to get @davwan01 and @sudpau01 on boarded to our development workflow.

wjwwood commented on rcl/include/rcl/time.h:
```diff
@@ -323,7 +323,7 @@ rcl_fini_duration(rcl_duration_t * duration);
 RCL_WARN_UNUSED
 rcl_time_source_t *
-rcl_get_default_ros_time_source(void);
+rcl_default_ros_time_source_get(void);
```
I think this one should be left as-is. Can you undo this particular name change?

wjwwood replied:
In fact all of the "`rcl_get_default_`..." function should be left alone I think. I'll try to comment on each of them below (at least in this header file).

You can think of it as a C++ function and it's clearer, i.e. this one would be:

```c++
rcl_time_source_t * rcl::get_default_ros_time_source()
```

You could also imagine it being `rcl_ros_time_source_get_default()` which in C++ would be like:

```c++
rcl_time_source_t * rcl::ros_time_source::get_default()
```

Where `rcl::ros_time_source` would be a class and `get_default` would be a static member function. But since both are equally valid, I'd prefer to leave it as-is.

davwan01 replied:
done

wjwwood commented on rcl/include/rcl/time.h:
```diff
@@ -340,7 +340,7 @@ rcl_get_default_ros_time_source(void);
 RCL_WARN_UNUSED
 rcl_time_source_t *
-rcl_get_default_steady_time_source(void);
+rcl_default_steady_time_source_get(void);
```
Same for this one.

davwan01 replied:
done

wjwwood commented on rcl/include/rcl/time.h:
```diff
@@ -357,7 +357,7 @@ rcl_get_default_steady_time_source(void);
 RCL_WARN_UNUSED
 rcl_time_source_t *
-rcl_get_default_system_time_source(void);
+rcl_default_system_time_source_get(void);
```
And here.

davwan01 replied:
done

wjwwood commented on rcl/include/rcl/time.h:
```diff
@@ -377,7 +377,7 @@ rcl_get_default_system_time_source(void);
 RCL_WARN_UNUSED
 rcl_ret_t
-rcl_set_default_ros_time_source(rcl_time_source_t * process_time_source);
+rcl_default_ros_time_source_set(rcl_time_source_t * process_time_source);
```
And this one.

davwan01 replied:
done

wjwwood commented on rcl/include/rcl/time.h:
```diff
@@ -431,7 +431,7 @@ rcl_get_time_point_now(rcl_time_point_t * time_point);
 RCL_WARN_UNUSED
 rcl_ret_t
-rcl_enable_ros_time_override(rcl_time_source_t * time_source);
+rcl_ros_time_enable_override(rcl_time_source_t * time_source);
```
I think I was wrong about this one. I believe it should be left as-is, because in this case the `ros_time` of `rcl_enable_ros_time_override` is referring to the concept of "ROS Time" not the `rcl_ros_time_source_t` C type.

Please revert this one.

wjwwood replied:
To be clearer, I would imagine the C++ version of this to be `rcl::enable_ros_time_override()` rather than `rcl::ros_time_source_t::enable_override()`.

davwan01 replied:
done

dirk-thomas commented:
> FYI this would affect your time based logging filter I think. It should just be a find-replace, sorry for the inconvenience.

No worry. The logging branch will be trivial to rebase once this is merged.

wjwwood commented on rcl/include/rcl/time.h:
```diff
@@ -447,7 +447,7 @@ rcl_enable_ros_time_override(rcl_time_source_t * time_source);
 RCL_WARN_UNUSED
 rcl_ret_t
-rcl_disable_ros_time_override(rcl_time_source_t * time_source);
+rcl_ros_time_disable_override(rcl_time_source_t * time_source);
```
Same thing for this one, revert it because `ros_time` is a concept not a type as I had originally thought.

davwan01 replied:
done

wjwwood commented on rcl/include/rcl/time.h:
```diff
@@ -465,7 +465,7 @@ rcl_disable_ros_time_override(rcl_time_source_t * time_source);
 RCL_WARN_UNUSED
 rcl_ret_t
-rcl_is_enabled_ros_time_override(rcl_time_source_t * time_source,
+rcl_ros_time_override_is_enabled(rcl_time_source_t * time_source,
```
Same for this as `rcl_disable_ros_time_override()`. Revert.

davwan01 replied:
done

wjwwood commented on rcl/include/rcl/time.h:
```diff
@@ -484,7 +484,7 @@ rcl_is_enabled_ros_time_override(rcl_time_source_t * time_source,
 RCL_WARN_UNUSED
 rcl_ret_t
-rcl_set_ros_time_override(rcl_time_source_t * time_source,
+rcl_ros_time_set_override(rcl_time_source_t * time_source,
```
Again here too, revert.

davwan01 replied:
done

davwan01 commented:
Hi @wjwwoodm,
I just updated my local patch and `push -f origin rename`. And updated the CI status Icon.
Is this correct sequence for updating the PR which has changes requested?
Thanks.

wjwwood commented:
[![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=2669)](http://ci.ros2.org/job/ci_windows/2669/)

davwan01 commented:
Thanks, @wjwwood :)

wjwwood commented:
Thank you @davwan01 😄.

@dirk-thomas FYI
